azurelinuxagent/pa/provision/default.py

```diff
@@ -166,11 +193,20 @@ def get_ssh_host_key_thumbprint(self, chk_err=True):
     def provisioned_file_path(self):
         return os.path.join(conf.get_lib_dir(), PROVISIONED_FILE)
 
+    def signaled_file_path(self):
+        return os.path.join(conf.get_lib_dir(), SIGNALED_FILE)
+
+    def isfile_provisioned(self):
+        return os.path.isfile(self.provisioned_file_path())
+
+    def isfile_signaled(self):
+        return os.path.isfile(self.signaled_file_path())
+
     def is_provisioned(self):
         '''
-        A VM is considered provisionend *anytime* the provisioning
+        A VM is considered provisioned *anytime* the provisioning
         sentinel file exists and not provisioned *anytime* the file
```

Comment:
See my previous comment about handle_provision_guest_agent(). I don't think the code, as changed, lives up to this correct comment.